src/t_zset.c

```diff
@@ -296,17 +314,15 @@ int zslDelete(zskiplist *zsl, double score, sds ele, zskiplistNode **node) {
  * element, which is more costly.
  *
  * The function returns the updated element skiplist node pointer. */
-zskiplistNode *zslUpdateScore(zskiplist *zsl, double curscore, sds ele, double newscore) {
+static zskiplistNode *zslUpdateScore(zskiplist *zsl, double curscore, sds ele, double newscore) {
     zskiplistNode *update[ZSKIPLIST_MAXLEVEL], *x;
     int i;
 
     /* We need to seek to element to update to start: this is useful anyway,
      * we'll have to update or remove it. */
     x = zsl->header;
     for (i = zsl->level - 1; i >= 0; i--) {
-        while (x->level[i].forward &&
-               (x->level[i].forward->score < curscore ||
-                (x->level[i].forward->score == curscore && sdscmp(x->level[i].forward->ele, ele) < 0))) {
+        while (compareNodeScoreEle(x->level[i].forward, curscore, ele) < 0) {
```

Comment:
I think it would make much more sense to pass the node to this function. the only case we are calling this is when we use zadd in order to change an element's score. when we have a direct pointer to the node we can just skip this search.

Reply:
Oh. Sorry not true, ignore. Forgot we must scan to take the update path in order to delete the node.

Reply:
If it's separate and non-trivial, let's do it as follow-up?